src/main/java/com/mathworks/ci/RunMatlabTestsStep.java

```diff
@@ -0,0 +1,185 @@
+package com.mathworks.ci;
+/**
+ * Copyright 2020 The MathWorks, Inc.
+ *  
+ */
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Copyright 2020 The MathWorks, Inc.
+ *  
+ */
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FilenameUtils;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import com.google.common.collect.ImmutableSet;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+public class RunMatlabTestsStep extends Step {
+    
+    private String testResultsPDF;
```

Comment:
We keep repeating code/data related to the artifacts in multiple places. It would be good eventually refactor this such that we don't keep repeating this. If we have to add a new artifact, think of all the places where changes have to be made.

Reply:
Sure , i get the point i shall invest some to see how we could achieve this to make it more easily extensible.